cpp/src/arrow/csv/writer.cc

```diff
@@ -234,7 +235,7 @@ class UnquotedColumnPopulator : public ColumnPopulator {
     while (offset < buffer_size) {
       // error happened or remaining bytes to check
       const char c = static_cast<char>(data[offset]);
-      if (c == '\n' || c == '\r' || c == ',' || c == '"') {
+      if (c == '\n' || c == '\r' || c == ',' || c == '"' || c == delimiter) {
```

Comment:
The comma can be removed here:
```suggestion
      if (c == '\n' || c == '\r' || c == '"' || c == delimiter) {
```

Reply:
Made the change, removed comma from check.